src/System.Collections/src/System/Collections/Generic/LinkedList.cs

```diff
@@ -186,9 +186,9 @@ public void CopyTo(T[] array, int index)
                 throw new ArgumentNullException("array");
             }
 
-            if (index < 0 || index > array.Length)
+            if ((uint)index > (uint)array.Length)
             {
-                throw new ArgumentOutOfRangeException("index", SR.Format(SR.IndexOutOfRange, index));
+                throw new ArgumentOutOfRangeException("index", index, SR.ArgumentOutOfRange_Index);
```

Comment:
This doesn't look valid, either.  ArgumentOutOfRange_Index says the index must be less than the size, but it's valid here for it to be equal to the size.